Fred2/CleavagePrediction/__init__.py

```diff
@@ -19,27 +19,57 @@
     pass
 
 
-class CleavageSitePredictorFactory(object):
-    class __metaclass__(type):
-        def __init__(cls, name, bases, nmspc):
-            type.__init__(cls, name, bases, nmspc)
-
-        def __call__(self, _predictor, *args, **kwargs):
-            """
-            If a third person wants to write a new Cleavage Site Predictor. One has to name the file fred_plugin and
-            inherit from ACleavagePrediction. That's it nothing more.
-            """
-
-            version = str(kwargs["version"]).lower() if "version" in kwargs else None
-            try:
-                return ACleavageSitePrediction[str(_predictor).lower(), version](*args)
-            except KeyError as e:
-                if version is None:
-                    raise ValueError("Predictor %s is not known. Please verify that such an Predictor is "%_predictor +
-                                "supported by FRED2 and inherits ACleavageSitePrediction.")
-                else:
-                    raise ValueError("Predictor %s version %s is not known. Please verify that such an Predictor is "%(_predictor, version) +
-                                "supported by FRED2 and inherits ACleavageSitePrediction.")
+class metaclassCleavageSite(type):
+    def __init__(cls, name, bases, nmspc):
+        type.__init__(cls, name, bases, nmspc)
+
+    def __call__(self, _predictor, *args, **kwargs):
+        """
+        If a third person wants to write a new Cleavage Site Predictor. One has to name the file fred_plugin and
```

Comment:
https://stackoverflow.com/questions/3483718/when-to-inline-definitions-of-metaclass-in-python